src/py42/clients/devices.py

```diff
@@ -265,3 +268,30 @@ def get_agent_full_disk_access_state(self, guid):
                 :class:`py42.response.Py42Response`: A response containing settings information.
             """
         return self.get_agent_state(guid, u"fullDiskAccess")
+
+    def get_settings_manager(self, guid):
+        """Gets setting data for a device and returns a `DeviceSettingsManager` for the target device.
+
+        Args:
+            guid (int,str): The globally unique identifier of the device.
+
+        Returns:
+            :class:`py42.clients._settings_managers.DeviceSettingsManager`: A class to help manage device settings.
+        """
+        settings = self.get_by_guid(guid, incSettings=True)
+        return DeviceSettingsManager(settings.data)
+
+    def update_device_settings(self, settings_manager):
```

Comment:
Instead of passing the whole manager, is there an object that would make more sense? Like a `settings_update` object that a manager can create that has everything it needs?  This works fine, but just feels strange from an API perspective.

Reply:
The manager is really just the settings response dict at its core with helper attributes and a few other things to make modifying the nested values easier, and make it so the user doesn't need to know the structure of the dict/json at all.

Technically this function could take a regular dict of device settings, the only change needed would be to call `dict()` on the input arg instead of passing `settings_manager.data` to the PUT request.